src/irmin/import.ml

```diff
@@ -88,4 +140,20 @@ module Seq = struct
       match s () with Seq.Nil -> false | Seq.Cons (v, s) -> f v || aux s
     in
     aux s
+
+  let of_seq_rev l =
+    let rec loop acc s =
+      match s () with Seq.Nil -> acc | Seq.Cons (x, tl) -> loop (x :: acc) tl
+    in
+    loop [] l
+
+  let of_seq l =
```

Comment:
This looks a lot like the stdlib implementation https://github.com/ocaml/ocaml/blob/trunk/stdlib/list.ml#L584

Reply:
indeed! my plan is to try with a smaller `direct_depth_default_` to see if that makes the s390s happy